todo-app/src/Components/AddTodoItem.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+
+import Button from "./Button";
+import InputField from "./InputField";
+import "./styles/addTodoItem.css";
+
+export default class AddTodoItem extends Component {
+  state = {
+    active: false,
+    toBeAddedItem: "",
+    toBeAddedDeadline: ""
+  };
+
+  toggleAddForm = event => {
+    this.setState({
+      active: !this.state.active
+    });
+
+    this.state.active && this.clearForm();
+  };
+
+  handleChangeTodoItem = event => {
+    this.setState({ toBeAddedItem: event.target.value });
+  };
+
+  handleChangeDeadline = event => {
+    this.setState({ toBeAddedDeadline: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+  };
+
+  clearForm = () => {
+    this.setState({
+      toBeAddedItem: "",
+      toBeAddedDeadline: ""
+    });
+  };
+
+  handleAddTodo = () => {
+    const { toBeAddedItem, toBeAddedDeadline } = this.state;
+    this.props.handleAddTodo(toBeAddedItem, toBeAddedDeadline);
+    this.setState({ active: false });
+    this.clearForm();
+  };
+
+  showForm = () => {
+    return (
+      <div className="addTodoForm">
+        <form onSubmit={this.handleSubmit}>
+          <div className="formInputWrapper">
+            <InputField
+              type="text"
+              placeholder="Todo item"
+              className="todoInput"
+              value={this.state.toBeAddedItem}
+              handleChange={this.handleChangeTodoItem}
+            />
+            <InputField
+              type="text"
+              placeholder="Deadline"
+              className="deadlineInput"
+              value={this.state.toBeAddedDeadline}
+              handleChange={this.handleChangeDeadline}
+            />
+          </div>
+          <div className="formButtonsWrapper">
+            <Button
+              className="addTodoButton"
+              buttonContent="Add"
+              onClick={this.handleAddTodo}
+              isDisabled={this.state.toBeAddedItem.trim() === ""}
+            />
+            <Button
+              className="cancelAddButton"
+              buttonContent="Cancel"
+              onClick={this.toggleAddForm}
+            />
+          </div>
+        </form>
+      </div>
+    );
+  };
+
+  render() {
+    return (
+      <div>
+        {!this.state.active && (
+          <button className="toggleAddForm" onClick={this.toggleAddForm}>
```

Comment:
This is more a personal preference, but when you use class names in React, it's a good practice to prefix it with the name of the component, e.g. `"AddTodoItem-toggleAddForm"`